### PR TITLE
fix: add /copy code command to extract clean code blocks without line numbers

### LINKED ISSUE #1388: Read-only mode copies line numbers with code, making copy-paste unusable
### What happened?

When using Qwen CLI in read-only mode, code blocks are rendered with line numbers.
When copying the code, the line numbers and separator characters (│) are also copied.

This makes pasted code invalid and requires manual cleanup line-by-line.

- Steps to Reproduce
1. Run Qwen CLI in read-only mode
2. Ask for code generation
3. Copy the generated code
4. Paste into an editor

-Actual Behavior
Copied text includes line numbers and separators:
│    69 ...
│    70 ...

- Impact
This significantly hurts developer productivity and copy-paste workflows.

-Suggested Solutions
- Add an option to disable line numbers
- Strip line numbers on copy
- Provide a "copy clean code" mode or flag

### What did you expect to happen?

Copied code should be clean, without line numbers or UI decorations.

### Client information

<details>
<summary>Client Information</summary>

CLI Version           0.6.0                                      │
│ Git Commit            17eb20c1                                   │
│ Model                 coder-model                                │
│ Sandbox               no sandbox                                 │
│ OS Platform           linux                                      │
│ OS Arch               x64                                        │
│ OS Release            6.14.0-37-generic                          │
│ Node.js Version       v24.11.1                                   │
│ NPM Version           11.6.2                                     │
│ Session ID            c840cc7f-d085-4065-a1d9-13b1e93ad676       │
│ Auth Method           qwen-oauth                                 │
│ Memory Usage          339.0 MB                                   │
│ IDE Client            VS Code       
</details>

### Login information

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Br1an67**
## TLDR

Add a `/copy code` subcommand that extracts only fenced code block contents from the last AI response, providing clean code ready to paste into an editor — without line numbers or UI decorations.

Closes #1388

## Dive Deeper

When using read-only mode, code blocks are rendered with line numbers and separator characters (`│`). Terminal text selection inevitably includes these decorations, making pasted code invalid.

While `showLineNumbers` can be toggled off in settings, a more ergonomic solution is to offer `/copy code` that extracts raw code from fenced blocks in the markdown source — before any TUI rendering.

**Changes:**
- `copyCommand.ts`: Added `extractCodeBlocks()` helper using multiline regex to extract fenced code block contents. Extended the `/copy` action to accept an optional `code` argument.
- `copyCommand.test.ts`: Added 7 new tests covering `/copy code` behavior and `extractCodeBlocks()` edge cases.

The original `/copy` behavior (copying full output) remains unchanged when no argument is provided.

## Reviewer Test Plan

1. Ask the AI to generate code (e.g., "write a Python hello world")
2. Run `/copy code` — verify only the code block content is in clipboard, without language tags or fence markers
3. Run `/copy` — verify the full markdown output is still copied as before
4. Ask for a response with no code blocks, then run `/copy code` — verify the "No code blocks found" info message

## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |
| npx      | ❓  | ❓  | ❓  |

## Linked issues / bugs

Closes #1388

### CODE REVIEW

File: packages/cli/src/ui/commands/copyCommand.ts [hunk 1]
```diff
@@ -9,13 +9,27 @@ import type { SlashCommand, SlashCommandActionReturn } from './types.js';
 import { CommandKind } from './types.js';
 import { t } from '../../i18n/index.js';
 
+/**
+ * Extracts all fenced code blocks from a markdown string,
+ * returning only the code content without the fence markers or language tags.
+ */
+export function extractCodeBlocks(markdown: string): string[] {
+  const codeBlockRegex = /^```[^\n]*\n([\s\S]*?)^```/gm;
```
Comment (wenshao) [inline packages/cli/src/ui/commands/copyCommand.ts:17]: **[Suggestion]** `extractCodeBlocks()` only recognizes triple-backtick fences, so `/copy code` will miss valid fenced code blocks that use tildes (`~~~`) or longer backtick fences. That makes the feature unreliable for legitimate Markdown responses that need a different fence style.

```suggestion
export function extractCodeBlocks(markdown: string): string[] {
  const codeBlockRegex = /^(?<fence>`{3,}|~{3,})[^\n]*\n([\s\S]*?)^\k<fence>[ \t]*$/gm;
  const blocks: string[] = [];
  let match;
  while ((match = codeBlockRegex.exec(markdown)) !== null) {
    blocks.push(match[2].replace(/\n$/, ''));
  }
  return blocks;
}
```

_— gpt-5.4 via Qwen Code /review_

**Review by DragonnZhang — APPROVED** [review]
LGTM! ✅ Clean implementation with solid test coverage (7 new tests for /copy code + extractCodeBlocks). The only suggestion (tilde fence support) is non-blocking and can be a follow-up. Windows Node 20 CI failure appears unrelated to these platform-independent changes. _— Qwen Code /review_

**Comment by tanzhenxin** [comment]
Thanks for this, @Br1an67, and sorry for the slow review.

Since this was opened, the `/copy` command was reworked in #4761 (merged), and it now includes exactly this — `/copy code` extracts the raw fenced code-block contents from the last AI response, with no line numbers or `│` decorations (it reads the markdown source before TUI rendering). It also adds language/index selection (`/copy <lang>`, `/copy code N`) and LaTeX support.

Because that landed in main, this PR is now superseded and has also picked up merge conflicts in `copyCommand.ts`. Closing it out — but thank you for driving the original fix; the shipped behavior matches what you proposed here. 🙏
